src/Google/GlobalSiteTag.php

```diff
@@ -156,14 +195,241 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
 		$order->save_meta_data();
 
 		printf(
-			'<script>gtag("event", "conversion", {"send_to": "%s","value": "%s","currency": "%s","transaction_id": "%s"});</script>',
+			'<script>
+            gtag("event", "conversion", 
+                {"send_to": "%s",
+                 "value": "%s",
+                 "currency": "%s",
+                 "transaction_id": "%s"});
+            </script>',
 			esc_js( "{$ads_conversion_id}/{$ads_conversion_label}" ),
 			esc_js( $order->get_total() ),
 			esc_js( $order->get_currency() ),
 			esc_js( $order->get_id() ),
 		);
 	}
 
+	/**
+	 * Display the JavaScript code to track the product view page.
+	 */
+	public function display_view_item_event_snippet(): void {
+		// Only display on the product view page.
+		if ( ! is_product() ) {
+			return;
+		}
+		$product = wc_get_product( get_the_ID() );
+		printf(
+			'<script>
+                gtag("event", "view_item", {
+                    "send_to": "GLA",
+                    "developer_id.%s": "true",
+                    "ecomm_pagetype": "product",
+                    "value": "%s",
+                    items:[{
+                    "id": "gla_%s",
+                    "price": %s,
+                    "google_business_vertical": "retail",
+                    "name":"%s",
+                    "category":"%s",
+                    }]});
+			</script>',
+			esc_js( self::DEVELOPER_ID ),
+			esc_js( (string) $product->get_price() ),
+			esc_js( $product->get_id() ),
+			esc_js( (string) $product->get_price() ),
+			esc_js( $product->get_name() ),
+			esc_js( join( '& ', $product->get_categories() ) ),
+		);
+	}
+
+	/**
+	 * Display the JavaScript code to track all pages.
+	 */
+	public function display_page_view_event_snippet(): void {
+		printf(
+			'<script>
+                gtag(
+                    "event", "page_view", {
+                    "send_to": "GLA",
+                    "developer_id.%s": "true",});
+			</script>',
+			esc_js( self::DEVELOPER_ID )
+		);
+	}
+
+	/**
+	 * Display the JavaScript code to track the cart page.
+	 */
+	public function display_cart_page_snippet(): void {
+		// Only display on the cart page.
+		if ( ! is_cart() ) {
+			return;
+		}
+
+		$item_info = '';
+
+		foreach ( WC()->cart->get_cart() as $cart_item ) {
+			// gets the product id
+			$id = $cart_item['product_id'];
+
+			// gets the product object
+			$product = $cart_item['data'];
+			$name    = $product->get_name();
+			$price   = $product->get_price();
+			// gets the cart item quantity
+			$quantity = $cart_item['quantity'];
+
+			$item_info = $item_info . sprintf(
+				'{
+				"id": "gla_%s",
+				"price": %s,
+				"google_business_vertical": "retail",
+				"name":"%s",
+				"quanitity":"%s",
+				}',
+				esc_js( $id ),
+				esc_js( $price ),
+				esc_js( $name ),
+				esc_js( $quantity )
+			);
+		}
+		$value = WC()->cart->total;
+		printf(
+			'<script>
```

Comment:
We repeat printing script tags quite often. Since we now support a minimum of WordPress 5.7, why don't we use the function [wp_print_inline_script_tag](https://developer.wordpress.org/reference/functions/wp_print_inline_script_tag/) for this.